dwave/system/samplers/clique.py

```diff
@@ -126,11 +109,95 @@ def properties(self):
         return self.properties
 
     @property
-    def largest_clique_size(self):
-        """The maximum number of variables that can be embedded.
-        """
+    def largest_clique_size(self) -> int:
+        """The maximum number of variables that can be embedded."""
         return len(self.largest_clique())
 
+    @property
+    def qpu_linear_range(self) -> Tuple[float, float]:
+        """Range of values possible for the qubit biases (linear biases),
+        h, for the QPU.
+        """
+        try:
+            return self._qpu_linear_range
+        except AttributeError:
+            pass
+
+        # get the energy range
+        try:
+            energy_range = tuple(self.child.properties['h_range'])
+        except KeyError as err:
+            # for backwards compatibility with old software solvers
+            if self.child.solver.is_software:
+                energy_range = (-2, 2)
+            else:
+                raise err
+
+        self._qpu_linear_range = energy_range
+
+        return energy_range
+
+    @property
+    def qpu_quadratic_range(self) -> Tuple[float, float]:
```

Comment:
Should we not also have one for extended J range? It might also differ between QPUs.

Reply:
good catch, I had accidentally removed it